src/net_processing.cpp

```diff
@@ -3273,8 +3273,9 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
             // - protocol version per the VERSION message supports WTXID_RELAY;
             // - we intended to exchange transactions over this connection while establishing it
             //   and the peer indicated support for transaction relay in the VERSION message;
-            // - we are not in -blocksonly mode.
-            if (pfrom.m_relays_txs && !m_ignore_incoming_txs) {
+            // - we are not in -blocksonly mode;
+            // - this is not a feeler.
+            if (pfrom.m_relays_txs && !m_ignore_incoming_txs && !pfrom.IsFeelerConn()) {
```

Comment:
cfd5c27dcdc39cf7a65c56c3dbf3bf60ea0c475f:

There are several unrelated things:

* If this feeler check isn't dead code here, it should probably be moved to the previous section that initializes the TxRelay struct. (And maybe in a separate pull request, as it would change non-erlay code?) This would mean that the tx relay struct isn't initialized and also m_relays_txs isn't set, which is then checked in this line here.
* Unrelated, I wonder if `RejectIncomingTxs` should be used here. The difference is that the `Relay` permission will be carved out. I don't know the answer, I just wanted to raise the point.

Reply:
good point, i'll look closer.

Reply:
1. From my understanding, this code isn't dead. The feeler check could indeed be moved above. Not sure what's the best way to attack this w.r.t. this PR (surely, a separate PR is the best option, but is that gonna get merged sooner...)
2. Yeah checking for tx relay support is kinda ugly in general. `RejectIncomingTxs` is of course relevant, but not 100% match, if you just look at the name. I'd better keep the code as-is now, and refactor this separately (dunno how yet)

Reply:
By the latter I mean that even `RejectIncomingTxs`-ing (whatever that could mean) doesn't necessarily mean we should reject reconciling. Might be useful. Not 100% sure (certainly not how Erlay intends to work in the default case).

Reply:
> Not sure what's the best way to attack this w.r.t. this PR 

Happy to work on this, unless you want to

Reply:
I was waiting for another opinion to show up here :)
Feel free to open a separate PR with that stuff, I will ack. Or, if you think it makes more sense here,, then I can just add a commit here, that i can do.

Reply:
Heh, maybe it should be left as-is here. We also send a lot of other messages, such as `GETADDR`, so the `SENDTXRCNCL` message seems to be consistent. Changing the tx-relay allocation could still be done, though. See https://github.com/bitcoin/bitcoin/pull/26396

Reply:
#26396 seems likely to be accepted, so I agree that the commit can be removed here.